lib/nbp/push_connector.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+module Nbp
+  class PushConnector
+    include Singleton
+
+    class Error < StandardError; end
+    class SettingsError < Error; end
+    class ServerError < Error; end
+
+    def initialize
+      super
+      create_source! unless source_exists?
+    end
+
+    def self.enabled?
+      Settings.nbp&.push_connector&.enable || false
```

Comment:
Shall we memoize this value?

```suggestion
      @enabled ||= Settings.nbp&.push_connector&.enable || false
```

Reply:
Can't we or do we have issues with concurrency? 

```
@mutex = Mutex.new # in the initializer
@enabled ||= @mutex.synchronize { Settings.nbp&.push_connector&.enable || false }
```

Reply:
I did not want to deal with mutexes just for memorizing a variable but I think we don't even need mutexes here. Since all threads have the same Settings object I think we can use a normal variable because read accesses to the settings are not a problem and the writes to `@enabled` should always be identical so there is no problem with threads overwriting each other.

Reply:
This assumes, of course, that a variable getting the same value written to by two threads at the same time will always end up with that value and not break in some weird way.

Reply:
Alright, thanks. I think this should work correctly 👍